app/femr/business/services/system/InventoryService.java

```diff
@@ -217,8 +220,9 @@ public InventoryService(IRepository<IMedication> medicationRepository,
         IMedicationInventory medicationInventory;
         MedicationItem medicationItem;
         try {
-            //This should exist already, so no need to query for unique.
-            medicationInventory = medicationInventoryRepository.findOne(medicationInventoryExpressionList);
+            //This should exist already, so no need to query for unique, but
+            //Query for unique to make sure the same medication is not readded. It happened in testing
+            medicationInventory = medicationInventoryExpressionList.findUnique();
```

Comment:
@anakhleh shouldn't this be exactly the same? Inside the `medicationInventoryRepository` we are just calling `findUnique()` on the expression list